### DISCUSSION TITLE
Code samples for hooking up OrbitDB, Helia, libp2p and public IPFS network

### DISCUSSION THREAD

**Discussion by allen-mak** [orbitdb/orbit-db#1212] [General]
Hello, I am trying to setup OrbitDB as a decentralized database using public IPFS network as the storage.
Could you please advise the right configuration that?

Another question is how to get the CID and pin the OrbitDB storage? Thank you.

**Comment by haydenyoung** [comment]
I'm not sure I understand the question. OrbitDB uses Helia for block swapping (I.e. moving records across the network from one OrbitDB to another). See the IPFSStorage for how OrbitDB uses IPFS.

**Comment by KempfCreative** [comment]
I think the question here is a good one. The amount of documentation one has to read in order to understand the right configuration to get a basic POC running is _staggering_. I don't feel that the orbitdb/examples repo really covers a lot of the common use cases in a way that is satisfactory (e.g. creating a database, storing and accessing data from a public IPFS network, public routing between nodes, etc).

Some common use cases would go a long way in helping adoption of this technology IMHO.

**Comment by haydenyoung** [comment]
Moving this to discussions. This seems to be more a question about general usability rather than a software feature or bug.
